### PR TITLE
fix: normalize hue to [0, 360) in getHue()

### LINKED ISSUE #58039: ColorPicker: color with hue near 360° (e.g. #DB393A) renders as purple on canvas
### Reproduction link

[https://5x.ant.design/components/color-picker-cn](https://5x.ant.design/components/color-picker-cn)

### Steps to reproduce

1. Pass value="#DB393A" to <ColorPicker />
2. Open the picker panel
3. The canvas selector dot and color preview show purple/magenta instead of red

### What is expected?

Display red (#DB393A)

### What is actually happening?

Display purple/magenta (~#DB39DB)

| Environment | Info |
| --- | --- |
| antd | 5.25.3 |
| React | 18.xmacOS |
| System | macOS |
| Browser | 谷歌 130  |

---
In @ant-design/fast-color, getHue() computes:
  round(60 × ((57−58)/162   6)) = round(359.63) = 360

360 is out of the valid range [0, 360). In fromHsv(), h=360 gives i=6,
which falls to default (case 5, magenta sector), boosting the blue channel
from 58 to 219, resulting in purple.

Fix: add % 360 in getHue() → this._h = round(60 * (...)) % 360

### PR DESCRIPTION
**Pull request by waterWang**
## Summary

When `getHue()` computes the hue for a color near the red end of the spectrum (e.g. `#DB393A`), the `round()` call can return 360, which is outside the valid hue range [0, 360). This causes the ColorPicker to render the wrong color (purple/magenta instead of red).

## Root Cause

In `src/FastColor.ts`, the `getHue()` method computes:

```
round(60 × ((57−58)/162 + 6)) = round(359.63) = 360
```

When 360 is passed to `toHsv()` or `toHsl()`, the downstream conversion (e.g. in `@ant-design/colors` or the ColorPicker component) produces incorrect results because the hue value is out of range.

## Fix

Added `this._h %= 360` after the hue computation in `getHue()`, matching the normalization already done in `fromHsl()` and `fromHsv()`:

```typescript
this._h = ((_h % 360) + 360) % 360;  // fromHsl()
this._h = ((_h % 360) + 360) % 360;  // fromHsv()
this._h %= 360;  // getHue() — new
```

## Related

Closes ant-design/ant-design#58039

## Summary by CodeRabbit

* **Bug 修复**
  * 修复色相计算结果超出有效范围的问题，确保色相始终处于 0–359°。
  * 优化色相结果处理，提升相关颜色显示的一致性。

### CODE REVIEW

**Review by yoyo837 — COMMENTED** [review]
The fix looks correct, but it needs a regression test covering the RGB-to-hue path.

File: src/FastColor.ts [hunk 1]
```diff
@@ -225,6 +225,7 @@ export class FastColor {
               : (this.r - this.g) / delta + 4),
         );
       }
+      this._h %= 360;
```
Comment (yoyo837) [inline src/FastColor.ts:228]: Please add a regression test for the RGB-to-hue conversion path. The existing test suite also passes on the base commit, so it does not protect this fix. A focused case using `new FastColor('#DB393A')` should assert that `getHue()`, `toHsv().h`, and `toHsl().h` all return `0`; these assertions return `360` on the base commit and pass with this change.

**Comment by afc163** [comment]
Thanks for the fix. Please add a regression test covering the RGB-to-hue path, for example:

```ts
expect(new FastColor('#DB393A').getHue()).toBe(0);
```

This should ensure the original issue is covered and prevent the boundary case from regressing.
